### ISSUE TITLE
AMP4EMAIL: Setting Form Values from State in Templates + Inconsistent Behavior Between Gmail Playground and Client

### ISSUE THREAD

**Issue by avcohen** [ampproject/amphtml#25920]
## What's the issue?

Unable to dynamically set a a `form` `input` field's value via the `[value]` attribute while using `amp-bind` within an `amp-mustache` template. This issue only occurs in the Gmail client. It works as expected within the [Gmail Amp Playground](https://amp.gmail.dev/playground/) and within the browser.

## How do we reproduce the issue?
I've built a contrived example of the issue which can be seen below.

The gist is available [here](https://gist.github.com/avcohen/381d86374215a55d527a97f931ce973c) with client side code.

While it's likely not needed, here's the server-side logic:

```csharp
        public class TestRequestModel {
            public string TestId { get; set; }
        }
        private string ValidId = "correct";
        [HttpPost]
        public async Task<IActionResult> TestGetId([FromForm]AEGetRepresentativesRequestModel requestModel)
        {
            AddRequiredAmpHeaders();
            if (requestModel == null)
            {
                return BadRequest(new {});
            }
            return Ok(new TestRequestModel { TestId = ValidId });
        }
        [HttpPost]
        public async Task<IActionResult> TestUseId([FromForm]TestRequestModel requestModel)
        {
            AddRequiredAmpHeaders();
            if (requestModel == null)
            {
                return BadRequest(new {});
            }
            if (requestModel.TestId != ValidId)
            {
                return BadRequest(new { Error = $"Invaild id: {requestModel.TestId}" });
            }
            return Ok(new { TestId = ValidId });
        }
```

## What browsers are affected?

This occurs in Chrome, Firefox and Safari. I'm using OS X Catalina and was also tested on an Ubuntu machine.

## Which AMP version is affected?

AMP4Email - v0.js via CDN @ https://cdn.ampproject.org/v0.js

**Comment by avcohen** [comment]
Just to follow up I did post this on Stack Overflow as well and a Developer Advocate @ Google [noted](https://stackoverflow.com/questions/59221285/amp4email-setting-form-values-from-state-in-templates?noredirect=1#comment104676182_59221285) there is currently a bug with `event.response` within forms in the Gmail client. If anyone is aware of where this bug is documented I'd be curious to see!

**Comment by zhangsu** [comment]
/to @caroqliu 

It looks like `amp-form` isn't parsing the response correctly in the SSR path. With the following AMP HTML:

```
<!doctype html>
<html ⚡4email>
<head>
  <meta charset="utf-8">
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
  <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
  <style amp4email-boilerplate>body{visibility:hidden}</style>
</head>
<body>
  <p [text]="state">Not submitted</p>
  <form action-xhr="https://amp.dev/documentation/examples/api/hello" method="get" on="submit-success:AMP.setState({state: event.response })">
    <input type="hidden" name="name" value="John">
    <input type="submit">
  </form>
</body>
</html>
```
In Vanilla AMP, after the form submission, the div text becomes `[object Object]`. But in the SSR path, the div text becomes `{"message":"Hello, John!"}`, which looks like a JSON string. It should be parsed into a JSON object instead:

https://github.com/ampproject/amphtml/blob/a0f855807981b65fa662395e1ae30f59c68be3fa/extensions/amp-form/0.1/amp-form.js#L808
https://github.com/ampproject/amphtml/blob/a0f855807981b65fa662395e1ae30f59c68be3fa/extensions/amp-form/0.1/amp-form.js#L812

The `body` field contains a JSON string, which needs to be converted to a JSON object:

https://github.com/ampproject/amphtml/blob/a0f855807981b65fa662395e1ae30f59c68be3fa/src/utils/xhr-utils.js#L108

**Comment by avcohen** [comment]
Hello. Thanks again for everyone's work on this. Is there a timeline for this to be released? The issue still seems to persist in Gmail (unsure if there's another issue endemic to the Gmail client?)

**Comment by caroqliu** [comment]
According to the [release schedule](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md#determining-if-your-change-is-in-production), this change should be available in two Tuesdays, or January 14.

**Comment by zhangsu** [comment]
We expect the fix to be in Gmail ~1.5 week after it's in AMP prod, so ~Jan 24.
